### PR TITLE
Add DEX mean price endpoint to webserver

### PR DESCRIPTION
**Pull request by davidmisiak**
Expose the mean prices in the webserver, filtered by the DEX pool address and asset pairs.

This PR depends on https://github.com/dcSpark/carp/pull/119, so only the last commit is relevant.

Example http query:

```
POST http://localhost:3000/dex/mean-price
content-type: application/json

{
    "addresses": [
        "addr1z8nvjzjeydcn4atcd93aac8allvrpjn7pjr2qsweukpnaytvcg6zm2vds6mz9x3h3yalqmnf24w86m09n40q3tgqxjms9yu6v8"
    ],
    "assetPairs": [
        {
            "asset1": null,
            "asset2": {
                "policyId": "c0ee29a85b13209423b10447d3c2e6a50641a15c57770e27cb9d5073",
                "assetName": "57696e67526964657273"
            }
        }
    ],
    "limit": 3,
    "untilBlock": "c95cac3036fcea430836d3bd40905e36989a2ef0e3f0602bc69d3c2043b909d3"
}

```

### CODE REVIEW

File: webserver/server/app/controllers/DexMeanPriceController.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import { expectType } from 'tsd';
+import type { EndpointTypes } from '../../../shared/routes';
+import { Routes } from '../../../shared/routes';
+import { getAddressTypes } from '../models/utils';
+import type { DexMeanPriceResponse } from '../../../shared/models/DexMeanPrice';
+import { dexMeanPrices } from '../services/DexMeanPrice';
+
+const route = Routes.dexMeanPrice;
+
+@Route('dex/mean-price')
```
Comment (SebastienGllmt) [inline webserver/server/app/controllers/DexMeanPriceController.ts:19]: What is this query supposed to be for?

The use-case for this price feed is for wallets to have a webserver they can query to get the price of given assets

- Not clear why an address should be passed for this
- Not clear why a before/after tx should be passed for this
- Not clear why we're returning a list as a response for this

Can you clarify your vision for how a website would use this endpoint to show the latest price information about tokens?

The only thing I can think of is that this endpoint is meant as an intermediate step for some other price service that does the following:
1. Query the latest block
2. Query `dex/mean-price` for a list of known DEX addresses to find recent swaps (recent being defined as not too far from the best block queried in (1))
3. Aggregate these prices in the middleware according to some formula
4. Cache the aggregated result and return that
5. Invalidate the cache every X minutes and re-query from the database

Is this what you had in mind? If that's the case, I feel like using a materialized view is going to be significantly faster and more efficient than what we have here

Comment (davidmisiak) [reply webserver/server/app/controllers/DexMeanPriceController.ts:19]: The idea for this endpoint was to enable querying historical (and current) data - e.g. if you want to draw a price chart for a given DEX, asset pair and a time interval. I agree that it's not well suited for getting the latest price and a separate endpoint would be more appropriate. Do you think this endpoint won't be useful? What is the preferred next step - adding the latest price endpoint?

Comment (SebastienGllmt) [reply webserver/server/app/controllers/DexMeanPriceController.ts:19]: Okay, fair enough. So basically we want two endpoints then?

**Comment by miksax** [comment]
Do not merge. It is a part of https://github.com/dcSpark/carp/pull/125

**Comment by SebastienGllmt** [comment]
It's okay to close this now, right?

**Pull request closed without merging**